src/pages/Theme/index.tsx

```diff
@@ -1,13 +1,38 @@
+import styled from '@emotion/styled';
+import { useEffect,useState } from 'react';
 import { Navigate, useParams } from 'react-router-dom';
 
+import { fetchThemes } from '@/api/api';
 import { ThemeGoodsSection } from '@/components/features/Theme/ThemeGoodsSection';
 import { getCurrentTheme, ThemeHeroSection } from '@/components/features/Theme/ThemeHeroSection';
 import { RouterPath } from '@/routes/path';
-import { ThemeMockList } from '@/types/mock';
+import type { ThemeData } from '@/types';
+
 
 export const ThemePage = () => {
   const { themeKey = '' } = useParams<{ themeKey: string }>();
-  const currentTheme = getCurrentTheme(themeKey, ThemeMockList);
+  const [currentTheme, setCurrentTheme] = useState<ThemeData | null>(null)
+  const [isLoading, setIsLoading] = useState(true)
+
+  useEffect(() => {
+    const getTheme = async () => {
```

Comment:
```suggestion
    setIsLoading(true);
    const getTheme = async () => {
```

지금처럼 mount 할 때 데이터 요청을 한다면 15번 라인에 로딩 상태 초기값을 true로만 전달해두고, 비동기 요청 전에 isLoading을 true로 변경해주지 않아도 되겠지만 기능 요구사항이 변화하거나 리팩토링 하면서 함수가 추출되어 사용되거나 등등 시간에 따라 코드가 변화할 때는 mount가 아닌 상황에서도 활용될 수 있기에 비동기 요청 전에 한 쌍으로 로딩 상태를 변화시키는걸 추천드려요.

Reply:
오 네 알겠습니다! step3 브랜치에서 반영하였습니다.